hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -489,6 +541,52 @@ private SignatureObject generateSignatureObject(final byte[] hash) {
 				.build();
 	}
 
+	private void createSidecarFile(
+			final SidecarFile.Builder sidecarFileBuilder,
+			final Instant firstTxnInstant,
+			final RecordStreamType.SidecarType sidecarType
+	) throws IOException, NoSuchAlgorithmException {
+		// create a file only if there are any sidecar records of this type for the current period
+		if (sidecarFileBuilder.getSidecarRecordsCount() > 0) {
+			final var sidecarFile = new File(generateSidecarFilePath(firstTxnInstant, sidecarType.getSidecarId()));
+			try (FileOutputStream stream = new FileOutputStream(sidecarFile, false);
+				 SerializableDataOutputStream dos = new SerializableDataOutputStream(new BufferedOutputStream(stream))
+			) {
+				// write contents of sidecar
+				dos.write(serialize(sidecarFileBuilder));
+
+				// make sure the whole sidecar is written to disk before continuing
+				// with calculating its hash and saving it as part of the SidecarMetadata
+				dos.flush();
+				stream.flush();
+				stream.getChannel().force(true);
+				stream.getFD().sync();
+
+				LOG.debug(OBJECT_STREAM_FILE.getMarker(),
+						"Sidecar sidecarFilePath written successfully {}", sidecarFile.getName());
+				recordStreamFileBuilder.addSidecars(createSidecarMetadata(sidecarFile, sidecarType));
+			}
+		}
+	}
+
+	private SidecarMetadata.Builder createSidecarMetadata(
+			final File sidecarFile,
+			final RecordStreamType.SidecarType sidecarType
+	) throws IOException, NoSuchAlgorithmException, IllegalStateException {
```

Comment:
Would it be ok to remove `IllegalStateException`? Or maybe it is it caught and handled somewhere?

Reply:
Nice catch, it was a leftover from an earlier version of the code - removed now.